src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -502,3 +508,68 @@ export const updateForm = (
     return transformMongoError(error)
   })
 }
+
+/**
+ * Updates form settings.
+ * @param originalForm The original form to update settings for
+ * @param body the subset of form settings to update
+ * @returns ok(updated form settings) on success
+ * @returns err(MalformedParametersError) if email update is attempted for an encrypt mode form
+ * @returns err(database errors) if db error is thrown during form setting update
+ */
+export const updateFormSettings = (
+  originalForm: IPopulatedForm,
+  body: SettingsUpdateDto,
+): ResultAsync<
+  FormSettings,
+  | FormNotFoundError
+  | DatabaseError
+  | DatabaseValidationError
+  | DatabaseConflictError
+  | DatabasePayloadSizeError
+  | MalformedParametersError
+> => {
+  // If non-empty string or even if any array is passed in, return with error if
+  // form is encrypt mode.
+  if (originalForm.responseMode === ResponseMode.Encrypt && !!body.emails) {
```

Comment:
is this actually possible, since the  `emails` property in are discriminated within `EmailFormSchema`?

Reply:
It will just pass silently without setting the emails. Do we not want to respond with an error code in that case?

Reply:
I think if you wanted to prevent this, then it should be performed at the Joi validator (that's why I suggested always including `responseMode` in the DTO to differentiate). Silent failure isn't great, but I think there's also no harm either.

Reply:
The real question is are there actually two DTO types, to distinguish between encrypt and email?

Reply:
I'll let it pass silently then. Not really a failure since you shouldn't be giving emails to an encrypt form :P